[docs] note form prop update default behavior (#6776)
sveltejs/kit: documentation/docs/06-form-actions.md

Without an argument, `use:enhance` will emulate the browser-native behaviour, just without the full-page reloads. It will:

> - update the `form` property if the action belongs to the page and invalidate all data on a successful response
> - update the `form` property on a invalid response if the action belongs to the page

Reviewer: I think that including the warning below is probably enough. I think it makes it harder to understand these bullet points to include more detail here
```suggestion
- update the `form` property and invalidate all data on a successful response
- update the `form` property on a invalid response
```

Author: You're probably right - thought the same when writing it, you confirming it makes me confident removing it again

- update `$page.status` on a successful or invalid response
- call `goto` on a redirect response
- render the nearest `+error` boundary if an error occurs

> > By default the `form` property is only updated for actions that belong to the page because in the native form submission case you would be redirected to the page the action is on

Reviewer: Maybe we could be more specific about what "actions that belong to the page" means?
```suggestion
> By default the `form` property is only updated for actions that are in a `+page.server.js` alongside the `+page.svelte` because in the native form submission case you would be redirected to the page the action is on
```

- call `goto` on a redirect response
- render the nearest `+error` boundary if an error occurs

> > By default the `form` property is only updated for actions that are in `+page.server.js` alongside `+page.svelte` because in the native form submission case you would be redirected to the page the action is on

Reviewer: missing some articles
```suggestion
> By default the `form` property is only updated for actions that are in a `+page.server.js` alongside the `+page.svelte` because in the native form submission case you would be redirected to the page the action is on
```
